include/thingset.h

```diff
@@ -1566,6 +1579,12 @@ struct thingset_context
      */
     const uint8_t *msg_payload;
 
+    /**
+     * Payload may contain an update for a single element in an array,
+     * so any two-element arrays should be interpreted as such.
+     */
+    bool elementwise_array_updates;
```

Comment:
Through which API would this variable be set?

Nits: I suggest to place this behind `auth_flags` to avoid wasting memory because of padding. (now it's between two 32-bit pointers and the compiler would add 3 padding bytes behind the bool)

Reply:
> The idea is to encode the element index and the value as a two-element array, right?

Yes.

> Does this need a change in the ThingSet protocol spec?

I was trying to avoid a significant change to the spec, though I realise some sort of update will be necessary in the section on CAN reports, at least just to say that this is supported.

> And how would the receiving side know that the value should be treated as an index + a value and not a two-element array?

Using:

`bool elementwise_array_updates;`

e.g.
```c
thingset_context ts;

void import_record(uint8_t data, size_t len, bool is_report)
{
   ts.elementwise_array_updates = is_report;
   int err = thingset_import_record(&ts, data, len, &records_endpoint,
                                    THINGSET_BIN_IDS_VALUES);
   ts.elementwise_array_updates = false;
}
```

I desperately wanted to avoid this, but I couldn't see any other way of telling the decoder that 2-element arrays should be special-cased. I wondered about using a tag, but any available tag is going to take up too many bytes.